bin/git-release

```diff
@@ -88,7 +88,7 @@ if test $# -gt 0; then
   if [ "$need_changelog" = true ]; then
     git-changelog -t "$version"
   fi
-  git commit -a -m "$msg"
+  git commit -a -m "$msg" --allow-empty
```

Comment:
I wonder if this really a reasonable fix to this problem. I mean, do we _really_ want an empty commit in these cases? Wouldn't it make more sense to just ignore committing?

Reply:
@perlun 
IMHO, if someone keeps using the `git release` to make new release, she will expect to see each release has its corresponding commit, with special commit message. If a commit could be ignored, this assumption could not work any more. Especially, when the user specifies her custom commit message with `git release`, she might consider it as a bug if the message is missing in `git log` because the committing was ignored.

Reply:
That's a valid point. But still, it feels odd to have all-empty commits in a git tree.

My typical use case: I use `git release` for a shorthand for `git tag foo && git push origin foo`. Sometimes, there are modified files (like for Ruby gems where I typically bump the version in `version.rb`), but there are also cases when no files need to be modified just for the sake of making the release.

Reply:
Well, an empty commit is not quite odd.

1. If you use `merge --no-ff` to merge a fast-forward branch, you will get an empty merge commit at the top of git tree. Some teams require all merge operations should be done with `--no-ff` option. (I have been worked in one of these teams)
2. Some people suggest to use an empty commit as the initial commit of a git repo. For example, see https://bit-booster.com/doing-git-wrong/2017/01/02/git-init-empty/. At least, it proves that making an empty commit is not a taboo.

Reply:
> If you use merge --no-ff to merge a fast-forward branch, you will get an empty merge commit at the top of git tree. Some teams require all merge operations should be done with --no-ff option. (I have been worked in one of these teams)

Yeah, I understand - I personally hate these commits, but that's a different story. 😉 I know that this is the policy at certain places, and I respect that people have different opinions on the matter.

> Some people suggest to use an empty commit as the initial commit of a git repo. For example, see https://bit-booster.com/doing-git-wrong/2017/01/02/git-init-empty/.

That's an interesting point actually - it makes a certain bit of sense (haven't read the link.)

Reply:
@perlun 
What about adding an extra option called `--no-empty-commit`? So that you could use this option to avoid creating empty commit.

Reply:
Makes sense, thanks for adding!